synapse/handlers/appservice.py

```diff
@@ -26,15 +26,22 @@
 logger = logging.getLogger(__name__)
 
 
+def log_failure(failure):
+    logger.error("Application Services Failure: %s", failure.value)
+    logger.error(failure.getTraceback())
```

Comment:
We probably want to use `exc_info` here. See: https://github.com/matrix-org/synapse/blob/v0.8.0/synapse/notifier.py#L220